libpod/pod_api.go

```diff
@@ -195,6 +200,49 @@ func (p *Pod) StopWithTimeout(ctx context.Context, cleanup bool, timeout int) (m
 	return nil, nil
 }
 
+// StopInfraIfNeeded stops the pod's infra container unless there are
+// containers that haven't been removed, stopped or already exited.
+func (p *Pod) stopInfraIfNeeded(ctx context.Context) error {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	if !p.HasInfraContainer() {
+		return nil
+	}
+
+	infra, err := p.infraContainer()
+	if err != nil {
+		return err
+	}
+
+	allCtrs, err := p.runtime.state.PodContainers(p)
+	if err != nil {
+		return err
+	}
+
+	for _, ctr := range allCtrs {
+		if ctr.ID() == infra.ID() {
+			// Skip the receiver (dead lock) and the infra container.
```

Comment:
```suggestion
```
From an earlier version

Reply:
What does this mean? I can't really parse this comment

Reply:
Does this skip the receiver or just infra? I ran into the same issue here where podman would deadlock when trying to stop both the container calling and infra...

Reply:
We need to skip checking the infra container here as it is still running; we'd otherwise return early below and not stop it.